### PR TITLE
boards: curie_ble: add board.cmake

### PR DESCRIPTION
**Pull request by jfischer-no**
Add board.cmake for curie_ble board and allow to flash
image using dfu-util.

### CODE REVIEW

File: boards/arm/curie_ble/board.cmake [hunk 1]
```diff
@@ -0,0 +1,13 @@
+if(DEFINED ENV{ZEPHYR_FLASH_OVER_DFU})
+  set(BOARD_FLASH_RUNNER dfu-util)
+else()
+  set(BOARD_FLASH_RUNNER openocd)
+endif()
+
+set(BOARD_DEBUG_RUNNER openocd)
+
+board_runner_args(dfu-util "--pid=8087:0aba" "--alt=ble_core")
+board_runner_args(openocd --cmd-pre-load "targets 8" "--gdb-port=3334")
```
Comment (jfischer-phytec-iot) [inline boards/arm/curie_ble/board.cmake]: I'm not sure about this `"targets 8"`.

File: boards/arm/curie_ble/board.cmake [hunk 2]
```diff
@@ -0,0 +1,13 @@
+if(DEFINED ENV{ZEPHYR_FLASH_OVER_DFU})
```
Comment (mbolivar) [inline boards/arm/curie_ble/board.cmake]: This should be removed. We have some legacy environment variables, but those are for legacy purposes.

Nowadays, users can just run `cmake -DBOARD_FLASH_RUNNER=dfu-util` to set the default flasher, or run `west flash -r dfu-util` to pick a flash runner at runtime.

File: boards/arm/curie_ble/board.cmake [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+else()
+  set(BOARD_FLASH_RUNNER openocd)
+endif()
+
+set(BOARD_DEBUG_RUNNER openocd)
+
+board_runner_args(dfu-util "--pid=8087:0aba" "--alt=ble_core")
+board_runner_args(openocd --cmd-pre-load "targets 8" "--gdb-port=3334")
+
+include(${ZEPHYR_BASE}/boards/common/openocd.board.cmake)
```
Comment (mbolivar) [inline boards/arm/curie_ble/board.cmake]: Just use the include order of these files to set your defaults.

By including openocd.board.cmake first, you automatically set the flash and debug runners to `openocd`, unless a previous setting is already defined.

Users are free to override at cmake or west invocation time.

**Review by mbolivar — CHANGES_REQUESTED** [review]
I apologize for the lack of documentation on this subject in Zephyr's current docs tree, but there are a couple of changes needed.

I am working on runner and west documentation, but got sidetracked by a couple of other issues. I will make sure to point you at the docs PR when it arrives this week.

File: boards/arm/curie_ble/board.cmake (hunk 2 above)
Comment (jfischer-phytec-iot) [reply boards/arm/curie_ble/board.cmake]: if I understand you correctly, lines L:1...L:5 should be removed?

File: boards/arm/curie_ble/board.cmake (hunk 3 above)
Comment (jfischer-phytec-iot) [reply boards/arm/curie_ble/board.cmake]: understand, but it should be consistent, `arc/arduino_101_sss/board.cmake` and `x86/arduino_101/board.cmake` are already different

File: boards/arm/curie_ble/board.cmake (hunk 2 above)
Comment (mbolivar) [reply boards/arm/curie_ble/board.cmake]: Lines 1 through 8 inclusive should be removed. The arduino_101_sss and arduino_101 files have been grandfathered in from a time when we were using Kbuild and had a bunch of shell scripts to do the flashing; they're really not good examples.

**Comment by jfischer-phytec-iot** [comment]
@mbolivar ping

**Review by nashif — APPROVED** [review]

**Pull request merged**